flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointStatistics.java

```diff
@@ -438,4 +453,58 @@ public int hashCode() {
 			return Objects.hash(super.hashCode(), failureTimestamp, failureMessage);
 		}
 	}
+
+	/**
+	 * Statistics for a pending checkpoint.
+	 */
+	public static final class PendingCheckpointStatistics extends CheckpointStatistics {
```

Comment:
`@JsonSubTypes.Type(value = CheckpointStatistics.PendingCheckpointStatistics.class, name = "in_progress")` should be added to `CheckpointStatistics` otherwise deserialization will fail.